downloadDirectory/src/main/java/org/reactome/release/downloadDirectory/FetchTestReactomeOntologyFiles.java

```diff
@@ -0,0 +1,127 @@
+package org.reactome.release.downloadDirectory;
+
+import java.io.BufferedReader;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.PrintWriter;
+import java.io.UnsupportedEncodingException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.nio.file.StandardCopyOption;
+import java.nio.file.StandardOpenOption;
+import java.sql.Blob;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.gk.persistence.MySQLAdaptor;
+
+public class FetchTestReactomeOntologyFiles {
+	private static final Logger logger = LogManager.getLogger();
+	private static Connection connect = null;
+	private static Statement statement = null;
+	private PreparedStatement preparedStatement = null;
+	private static ResultSet resultSet = null;
+	
+	public static void execute(MySQLAdaptor dba, String username, String password, String host, String database, String releaseNumber) throws SQLException, ClassNotFoundException, UnsupportedEncodingException, FileNotFoundException, IOException {
+		logger.info("Running FetchTestReactomeOntologyFiles...");
+		Class.forName("com.mysql.jdbc.Driver");
+		connect = DriverManager.getConnection("jdbc:mysql://" + host + "/" + database + "?" + "user=" + username + "&password=" + password);
+		statement = connect.createStatement();
+		resultSet = statement.executeQuery("SELECT ontology FROM Ontology");
+		// The returned value is a single blob composed of binary and text. The three files produced by this step (pprj, pins, pont) are found within this blob.
+		// A handful of regexes and conditional statements are used to handle this data and output the 3 files. 
+		
+		PrintWriter pprjWriter = new PrintWriter("reactome_data_model.pprj");
+		PrintWriter pontWriter = new PrintWriter("reactome_data_model.pont");
+		PrintWriter pinsWriter = new PrintWriter("reactome_data_model.pins");
+		
+		while (resultSet.next()) {
+			
+			Blob blob = resultSet.getBlob("ontology");
+			InputStream is = blob.getBinaryStream();
+			BufferedReader br = new BufferedReader(new InputStreamReader(is));
```

Comment:
Could remove temp `InputStream is` and call `blob.getBinaryStream()` directly

Reply:
I would actually argue that it should be broken up further as: 
```
InputStream is =  blob.getBinaryStream();
InputStreamReader isr = new InputStreamReader(is);
BufferedReader br = new BufferedReader(isr);
```
This makes it easier to debug. Nesting too many `new...` calls could make it trickier to figure out what went wrong when the stack trace tells you there was a NullPointerException on one line, and that one line has multiple nested constructor calls.

Reply:
It looks like `blob.getBinaryStream()` throws an SQLException and `new BufferedReader(isr) and `new InputStreamReader(is)`  don't throw any documented exceptions.  I'm okay with keeping them separate if we feel that's better, but the exceptions thrown should be distinct in a trace.